code/model/modelcollide.cpp

```diff
@@ -668,34 +723,72 @@ void model_collide_parse_bsp_tmappoly(bsp_collision_leaf *leaf, SCP_vector<model
 
 	Assert(tmap_num >= 0 && tmap_num < MAX_MODEL_TEXTURES);
 
-	verts = (model_tmap_vert *)(p+44);
+	verts = new model_tmap_vert[nv];
+
+	// Copy the verts manually since they aren't aligned with the struct
+	unpack_tmap_verts(&p[44], verts, nv);
 
 	leaf->tmap_num = (ubyte)tmap_num;
 	leaf->num_verts = (ubyte)nv;
 	leaf->vert_start = (int)vert_buffer->size();
 
-	vec3d *plane_pnt = vp(p+20);
-	float face_rad = fl(p+32);
 	vec3d *plane_norm = vp(p+8);
 
-	leaf->plane_pnt = *plane_pnt;
-	leaf->face_rad = face_rad;
 	leaf->plane_norm = *plane_norm;
 
 	for ( i = 0; i < nv; ++i ) {
 		vert_buffer->push_back(verts[i]);
 	}
 }
 
+void model_collide_parse_bsp_tmap2poly(bsp_collision_leaf* leaf, SCP_vector<model_tmap_vert>* vert_buffer, void* model_ptr)
+{
+	ubyte* p = (ubyte*)model_ptr;
+
+	uint i;
+	uint nv;
+	model_tmap_vert* verts;
+
+	//Assert(Mc_pm->version >= 2300);
```

Comment:
Whoops, forgot to remove this. (Unless we have a better way of checking it)

Reply:
I couldn't find one honestly. The global pointers used in the model code honestly need to be cleaned up at some point to make version checking easier.